### PR TITLE
WIP: Add rocket chat appliance

### LINKED ISSUE #8358: Create /appliance/rocket-chat
[Brief doc](https://docs.google.com/document/d/1cp8qGmKwpoLUmenlOHc6j6MILkiVLh9mEaqIv_VLZSU/edit?disco=AAAAIvZwPLg&ts=5f6cc323&usp_dm=true)

Create:

- [x] /appliance/rocket-chat
- [x] /appliance/rocket-chat/raspbery-pi2
- [x] /appliance/rocket-chat/raspbery-pi
- [x] /appliance/rocket-chat/intel-nuc

Add:
- [x] /appliance/portfolio

Update:
- [x] featured include, replace mosquitto

### PR DESCRIPTION
**Pull request by pmahnke**
## Done

- Add rocket chat appliance
- Update portfolio page to build dynamically
- Update /appliance featured appliances
- **Note - actual downloads and checksums are missing**

## QA

- Check out this feature branch
- Run the site using the command `./run serve` or `dotrun`
- View the site locally in your web browser at: 
    - http://0.0.0.0:8001/appliance
    - http://0.0.0.0:8001/appliance/portfolio
    - http://0.0.0.0:8001/appliance/rocket-chat
    - http://0.0.0.0:8001/appliance/rocket-chat/raspberry-pi2
    - http://0.0.0.0:8001/appliance/rocket-chat/raspberry-pi
    - http://0.0.0.0:8001/appliance/rocket-chat/pc
    - http://0.0.0.0:8001/appliance/rocket-chat/intel-nuc
- Run through the following [QA steps](https://canonical-web-and-design.github.io/practices/workflow/qa-steps.html)
- See that it matches the [brief](https://docs.google.com/document/d/1cp8qGmKwpoLUmenlOHc6j6MILkiVLh9mEaqIv_VLZSU/edit?ts=5f50ca4b)

## Issue / Card

Fixes #8358

### CODE REVIEW

File: appliances.yaml [hunk 1]
```diff
@@ -87,3 +93,18 @@ appliances:
...
       4: True
+  rocket-chat:
+    name: "Rocket.Chat"
+    appliance_name: "Rocket.Chat Server"
+    url_name: "rocket-chat"
+    alias: "rocket-chat"
+    logo: "https://assets.ubuntu.com/v1/0abeb03e-rocket.chat.svg"
+    port: 3000
+    iso_url:
+      raspberrypi: "https://cdimage.ubuntu.com/ubuntu-core/appliances/plexmediaserver-core18-pi.img.xz"
```
Comment (anthonydillon) [inline appliances.yaml:104]: These are the wrong iso links.

File: templates/appliance/rocket-chat/_next-steps.html [hunk 2]
```diff
@@ -0,0 +1,32 @@
+<section class="p-strip--light">
+  <div class="row u-equal-height">
+    <div class="col-6">
+      <h2>
+        Start using your {{ appliance.name }} Ubuntu Appliance
+      </h2>
+      <p>
+        Now you can go ahead and set everything up.  It’s usually best to wait 5-10 minutes here. Some snaps will need to refresh themselves and the appliance may need to automatically reboot. When you’re ready go to:
+      </p>
+      <pre><code>http://&lt;IP-of-your-appliance&gt;:3000/web</code></pre>
```
Comment (anthonydillon) [inline templates/appliance/rocket-chat/_next-steps.html:10]: Shouldn't this use the data from the yaml?

File: webapp/app.py [hunk 3]
```diff
@@ -300,9 +302,7 @@ def utility_processor():
 # Ceph docs
 ceph_docs = Docs(
     parser=DocParser(
-        api=discourse_api,
-        index_topic_id=17250,
-        url_prefix="/ceph/docs",
+        api=discourse_api, index_topic_id=17250, url_prefix="/ceph/docs",
```
Comment (anthonydillon) [inline webapp/app.py:305]: Do you have a line limit set on prettier?

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Few  comments in the code

File: appliances.yaml (hunk 1 above)
Comment (pmahnke) [reply appliances.yaml:104]: yes, I put that in the note.

File: templates/appliance/rocket-chat/_next-steps.html (hunk 2 above)
Comment (pmahnke) [reply templates/appliance/rocket-chat/_next-steps.html:10]: the entire text changes with each app, not sure it is worth it.

Comment (pmahnke) [reply templates/appliance/rocket-chat/_next-steps.html:10]: done and updated the copy doc

File: templates/appliance/rocket-chat/index.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+      title: "License: MIT"
+      url: "https://github.com/RocketChat/Rocket.Chat/blob/develop/LICENSE"
+  compliance:
+    1: "EU GDPR"
+  base: "core18"
+  published_date: "2020-09"
+  maintenance_date: "2030-09"
+  snaps:
+    1:
+      name: "Rocket.chat server"
```
[the remark this answers is not in the document]
Comment (pmahnke) [reply templates/appliance/rocket-chat/index.md:43]: done and updated the copy doc

File: webapp/app.py (hunk 3 above)
Comment (pmahnke) [reply webapp/app.py]: atom seems not to... vs code does.

File: templates/appliance/rocket-chat/_next-steps.html [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+      <p>
+        in another computer’s browser. Where you will find the initial configuration wizard.
+      </p>
+      <p>
+        Follow the instructions to start using your new Rocket.chat appliance.
```
[the remark this answers is not in the document]
Comment (pmahnke) [reply templates/appliance/rocket-chat/_next-steps.html]: Updated the code and copy doc

**Review by anthonydillon — APPROVED** [review]
LGTM

**Review by anthonydillon — CHANGES_REQUESTED** [review]
Infact, going to code -1 until the links are updated.

**Pull request closed without merging**